BatchV2: Added a demo showcase notebook

igor-planet opened the pull request:
I've added a notebook that showcases some newer features of BatchV2, namely geopackages, FeatureManifest & Execution databases.

I've also had to put this notebook under the skip list, as there is a requirement to have your own AWS bucket & credentials with access to those buckets.

As this is my first contribution, I'd appreciate any feedback on overall styling and maybe checking if the code-style is in line with this repository.

jonasViehweger commented:
Some comments from my side:

- When defining `request_body`, I think it would be clearer to point out where the geopackage is stored. Maybe extract out `features.s3.url` to a variable as was done for `AWS_IAM_ROLE_ARN`. Since this is the defining feature of BatchV2 I think it makes sense to draw extra attention to where that gpkg is stored. Might also make sense to switch both parts around, so the exploration of the gpkg first and then the definition of the request body. Maybe even with a s3 call to put geopackage on S3 in between. But that's up to you, I think both works.
- For the `FeatureManifest`, there's one cell which calls `gdf_manifest.head()` but the output is not available. I would be curious how this file differs from the input geopackage, without having to run the full notebook myself. If that cell could be run I think it would provide some benefit.
- For the execution SQLite: Is there any docs on fields and relations for that db? If so, linking that in the notebook would be helpful

Those are just minor points, overall I think it's a nice notebook!

igor-planet commented:
Hi, thank you for your review!

I've applied your suggestions:
1. extracted the path to the GPKG into a separate variable
2. flipped the sections of request body & exploring input geopackage
3. ran the missing output for FeatureManifest  gdf_manifest.head()
4. added links the appropriate sections in the official documentation (not yet published at that location) for the FeatureManifest & Execution DB files.

matthew-ballard commented:
hey this is really great, really well written and easy to follow.  I don't have AWS access so I didn't actually test the whole notebook, though

Only two really small comments:

1. is the API called BatchV2? or is it just the version 2 of the Batch Processing API? The docs seem to say the latter, so I am wondering if the folder would be better called `Batch-Processing-API` to be more 1:1 with the docs?
2. In the not distant future we will be phasing out the name "Sentinel Hub" - what do you think about moving this out of the `Sentinel-Hub-APIs` folder and putting in the top folder with all the other APIs? That way the URL hopefully won't change in the future if you refer to it from other places
